docs/design/server_side_charts.md

```diff
@@ -0,0 +1,44 @@
+# Supporting charts on the server side
+As discussed in #367, the server side components manager and resourcifier have not yet been refactored, as of this writing, to work with [the new chart format](docs/design/chart_format.md). Presumably, expandybird should not be affected.
+
+## Context
+In discussions about the refactoring on the Slack channel, a number of potential requirements were discussed that could complicate matters, such as:
+* supporting multiple templating technologies within a single chart.
+* supporting multiple templating technologies within a single dependency graph.
+* supporting the parameterization of charts, as opposed to templates.
+* supporting more than one top level template in a single chart.
+* supporting dependency definition and resolution mechanism at the chart level.
+
+It was also observed that there is strong motivation to complete the first pass of the helm/dm integration, using the existing DM templating engine, before adding new features or addressing more complex scenarios.
+
+## Assertions
+In view of this context, commentators asserted that:
+* It seems unlikely for a single chart to require more than one generation technology.
+* It seems unnecessary that we need to support more than one top level template per chart, if templates can nest, since any forest of top level templates required by the user can be implemented using a tree, where a single top level template calls the members of the forest.
+
+If we accept these assertions, then parameterizing a chart becomes the same thing as parameterizing a single top level template. This is an important result, because it means that:
+* we don't need to rationalize schemas, type systems or parameterization mechanisms across multiple technologies.
```

Comment:
FWIW it's not hard to do this, especially since it seems that just a handful of primitive types are needed (string, number, bool).  But accepted that doing nothing is even easier than doing an easy task :)